### DISCUSSION TITLE
How to use the Context / Query API from custom page

### DISCUSSION THREAD

**Discussion by bastiW** [keystonejs/keystone#7329] [Questions] [answered]
How can I access the context / query api from a custom page? 

Is there an inbuilt solution or do I need to use my own graphql react client library like apollo?

Like this:

```ts
import { PageContainer } from '@keystone-6/core/admin-ui/components';
import { Heading } from '@keystone-ui/core';


export default function CustomPage () {

    const user = await context.query.User.findOne({
        where: { id: '...' },
        query: 'id name posts { id title }',
    });

    return (
        <PageContainer header={<Heading type="h3">Custom Page</Heading>}>
            <h1>Hello {user.id}</h1>
        </PageContainer>
    )
}

```

**Comment by dcousens** [comment, marked as answer]
Hi @bastiW,

You can't use the `context.query` or `context.db` API from a custom page,  as the custom page is part of the frontend and those idioms require direct access to the Prisma database and ORM.

You will need to use a GraphQL library like Apollo,  but we do have an example of how to do this in [`examples-staging`](https://github.com/keystonejs/keystone/blob/158db2f236a184779e3fa3ff34f7690aec5ab4e3/examples-staging/basic/admin/pages/post/%5Bid%5D.tsx#L170-L187) (warning, this folder may move soon).

I'll try and return back to this discussion if that example moves.

**Comment by DiesIrae** [comment]
Thanks! And how can we get session info?


I tried using `useKeystone()` and I get the logged user id, but I don't find its other infos I have in the session object.
